exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/metric_exporter/__init__.py

```diff
@@ -219,27 +243,448 @@ def export(
+                                            "data_points": split_data_points,
+                                        },
+                                    }
+                                ]
+                            elif metric.HasField("gauge"):
+                                split_metrics = [
+                                    {
+                                        "name": metric.name,
+                                        "description": metric.description,
+                                        "unit": metric.unit,
+                                        "gauge": {
+                                            "data_points": split_data_points,
+                                        },
+                                    }
+                                ]
+                            elif metric.HasField("summary"):
+                                split_metrics = [
+                                    {
+                                        "name": metric.name,
+                                        "description": metric.description,
+                                        "unit": metric.unit,
+                                        "summary": {
+                                            "data_points": split_data_points,
+                                        },
+                                    }
+                                ]
+
+                            split_scope_metrics = [
+                                {
+                                    "scope": scope_metrics.scope,
+                                    "schema_url": scope_metrics.schema_url,
+                                    "metrics": split_metrics,
+                                }
+                            ]
+                            split_resource_metrics = [
+                                {
+                                    "resource": resource_metrics.resource,
+                                    "schema_url": resource_metrics.schema_url,
+                                    "scope_metrics": split_scope_metrics,
+                                }
+                            ]
+
+                    if not split_data_points:
+                        # If data_points is empty remove the whole metric
+                        split_metrics.pop()
+
+                if not split_metrics:
+                    # If metrics is empty remove the whole scope_metrics
+                    split_scope_metrics.pop()
+
+            if not split_scope_metrics:
+                # If scope_metrics is empty remove the whole resource_metrics
+                split_resource_metrics.pop()
+
+        if batch_size > 0:
+            yield pb2.MetricsData(
+                resource_metrics=self._get_split_resource_metrics_pb2(
+                    split_resource_metrics
+                )
+            )
+
+    def _get_split_resource_metrics_pb2(
```

Comment:
This helper is used to write actual pb2 objects before each `yield` back to the `export` caller.